### DISCUSSION TITLE
app.startup not work.

### DISCUSSION THREAD

**Discussion by Chise1** [fastapi/fastapi#6979] [Questions] [answered]
Whern I try to test my project,I found app.startup not work.

src.main.py

```Python
from fastapi import FastAPI

app = FastAPI()


@app.on_event("startup")
async def startup() -> None:
    print("start")


@app.on_event("shutdown")
async def shutdown() -> None:
    print('shutdown')
@app.get("/")
async def root():
    print("root")
    print("root")
    print("root")
    print("root")
    return {"message": "Tomato"}
```

```test_main.py
import pytest
from httpx import AsyncClient

from src.main import app


@pytest.mark.asyncio
async def test_root():
    async with AsyncClient(app=app, base_url="http://test") as ac:
        response = await ac.get("/")
    assert response.status_code == 200
    assert response.json() == {"message": "Tomato"}
```
When I perform `pytest -s`,I got:

```
tests/test_main.py root
root
root
root

```

file tree:

.
├── src
│   ├── __init__.py
│   └── main.py
└── tests
    ├── __init__.py
    └── test_main.py

**Comment by krishnardt** [comment, marked as answer]
https://fastapi.tiangolo.com/tutorial/debugging/

This reference will help you to understand the reason.
Here importing app doesnt mean starting the server.

**Comment by tiangolo** [comment]
Thanks for the help here @krishnardt ! 👏  🙇 

Thanks for reporting back and closing the issue @Chise1  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
